extension/oauth2clientcredentialsauthextension/config.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauth2clientcredentialsauthextension
```

Comment:
Similarly the package name, perhaps shorten to `oauth2clientauthextension` or `oauth2authextension`, depending on what we decide for the name in the config.

Reply:
Thanks @tigrannajaryan. This has been a point of discussion in the past and I am conflicted on the naming too -  OAuth2 supports 3 types of work flows  for clients of which clientcredentials is one of them -  `oauth2/clientcredentials` is how golang arranges its packages - from our collector perspective we will only use client credentials - don't see using other 2. Not sure how to go about it - 

Here are my possible namings - I can split the package into subpackages 

- auth/oauth2/clientcredentials  (this naming is in the style of the storage extensions present in the extension) 
- auth/oauth2clientcredentials  (this naming is in the style of the storage extensions present in the extension) 
- auth/oauth2ccextension 
- oauth2extension (assuming `clientcredentials` workflow is implicilty understood by the users)  

I personally like 1 but please advice. I will make changes accordingly

Reply:
@gramidt @jpkrohling  let me know what you think here.